bin/i18n/sync-all.rb

```diff
@@ -36,11 +36,53 @@
 IN_UP_BRANCH = "i18n-sync-in-up-#{Date.today.strftime('%m-%d')}".freeze
 DOWN_OUT_BRANCH = "i18n-sync-down-out-#{Date.today.strftime('%m-%d')}".freeze
 
-def parse_options
-  options = {}
-  opt_parser = OptionParser.new do |opts|
-    opts.banner = <<-USAGE
-  Usage: sync-codeorg-all [options]
+class I18nSync
+  def initialize(args)
+    @options = parse_options(args)
+  end
+
+  def run
+    if @options[:interactive]
+      checkout_staging
+      sync_in if I18nScriptUtils.should_i "sync in"
+      sync_up if I18nScriptUtils.should_i "sync up"
+      create_in_up_pr if @options[:with_pull_request]
+      sync_down if I18nScriptUtils.should_i "sync down"
+      sync_out if I18nScriptUtils.should_i "sync out"
+      create_down_out_pr if @options[:with_pull_request]
+      upload_i18n_stats if I18nScriptUtils.should_i "upload translation stats"
+      checkout_staging
+    elsif @options[:command]
+      case @options[:command]
+      when 'in'
+        puts "Pulling all updated source strings into i18n/locales/sources"
+        sync_in
+        if @options[:with_pull_request]
+          create_in_up_pr
+        end
+      when 'up'
+        puts "Uploading i18n/locales/sources to crowdin"
+        sync_up
+      when 'down'
+        puts "Downloading translations from crowdin into i18n/locales"
+        sync_down
+      when 'out'
+        puts "Distributing translations from i18n/locales out into codebase"
+        sync_out
+        if @options[:with_pull_request]
+          create_down_out_pr
+        end
+      end
+    end
+  end
+
+  private
+
+  def parse_options(args)
```

Comment:
Why did you decide to make 'parse_options(args)' a private method?

Reply:
anything that's not part of the interface the class presents to the outside world - ie, any method that's only called by the class on itself and not by anything "outside" the class - should arguably be private.